### PR TITLE
Simplify RobotMap

### PR DESCRIPTION
**Pull request by ryanlarkin**

### CODE REVIEW

**Comment by ounsworth** [comment]
@Stealth2994 is gonna have to review this one because I don't know anything about the RobotMap file.

**Review by cnnradams — APPROVED** [review]
A lot better than the old one, some more commenting on that hackey code and maybe some simplification from <T> T if possible for rookies. Good for now.

**Comment by ounsworth** [comment]
@Stealth2994, @eandr127  I think you two have a handle on this, so I'll butt out, but there's a big Merge button after you've approved. Unmerged green checkmarks are weird. There's a Request Changes button if you don't think it's ready for merging.

**Pull request merged**
